akka-http-server/src/main/scala/endpoints/akkahttp/routing/BasicAuthentication.scala

```diff
@@ -0,0 +1,36 @@
+package endpoints.akkahttp.routing
+
+import java.util.Base64
+
+import akka.http.scaladsl.model.headers.{Authorization, BasicHttpCredentials}
+import akka.http.scaladsl.model.{HttpHeader, HttpResponse, StatusCodes}
+import akka.http.scaladsl.server.Directives
+import endpoints.algebra
+import endpoints.algebra.BasicAuthentication.Credentials
+
+trait BasicAuthentication extends algebra.BasicAuthentication with Endpoints {
+
+  /**
+    * Extracts the credentials from the request headers.
+    * In case of absence of credentials rejects request
+    */
+  private[endpoints] lazy val basicAuthentication: RequestHeaders[Credentials] =
+    Directives.headerValue(extractCredentials)
```

Comment:
How is the request rejected if the header is not present?

That’s probably out of the scope of this PR but we should clearly define what the semantic of this method should be. The Play routing rejects the request with a 401 status code, I think we should be consistent with that.